### PR TITLE
Fix TensorFlow LSTM Masking Layer Issue

### LINKED ISSUE #18397: Masking layer doesn´t work on GPU
Using a Masking layer in an LSTM model and Tensorflow as backend when training on CPU runs without problem, howerver, when using GPU the following error is shown:

```
Using a symbolic `tf.Tensor` as a Python `bool` is not allowed: AutoGraph is disabled in this function. Try decorating it directly with @tf.function.

Arguments received by LSTM.call():
  • sequences=tf.Tensor(shape=(None, 30, 5), dtype=float32)
  • initial_state=None
  • mask=tf.Tensor(shape=(None, 30), dtype=bool)
  • training=None
```

Here is a simple example in Google Colab where you can reproduce the exact behavior. Just make sure the GPU option is selected in Runtime:
https://colab.research.google.com/drive/1Kv90SJo1kcYye1Yl0nRHihRX3De1fBDd?usp=sharing

Otherwise, here is the code:

```
samples, timesteps, features = 32, 30, 8
x_train = np.random.random([samples, timesteps, features]).astype(np.float32)
x_train[:, 3, :] = 0.
x_train[:, 5, :] = 0.

y_train = np.random.random([samples, 1]).astype(np.float32)
inputs = keras_core.layers.Input(shape=(timesteps, features,))
mask = keras_core.layers.Masking(mask_value=0.)(inputs)
lstm = keras_core.layers.LSTM(32)(mask)
dense = keras_core.layers.Dense(1)(lstm)
model = keras_core.Model(inputs, dense)

model.compile(optimizer=keras_core.optimizers.Adam(learning_rate=0.001), loss=keras_core.losses.MeanSquaredError())
model.fit(x_train, y_train, epochs=10, batch_size=32)
```

Thanks in advance

### PR DESCRIPTION
**Pull request by sampathweb**
Fixes #18397 by wrapping `lstm` and `gru` in `tf.function`.  Also decorated `_cudnn_lstm` to be aligned with `cudnn_gru`.

Failing JAX GPU test is from unrelated issue - 
keras/initializers/constant_initializers_test.py::ConstantInitializersTest::test_constant_initializer Fatal Python error: Aborted

### CODE REVIEW

**Review by fchollet — APPROVED** [review]
LGTM, thank you!
